wincode-derive/src/common.rs

```diff
@@ -307,6 +305,128 @@ impl Variant {
     }
 }
 
+pub(crate) trait VariantsExt {
+    /// Generate the `TYPE_META` implementation for an enum.
+    fn type_meta_impl(&self, trait_impl: TraitImpl, tag_encoding: &Type) -> TokenStream;
+}
+
+impl VariantsExt for &[Variant] {
+    fn type_meta_impl(&self, trait_impl: TraitImpl, tag_encoding: &Type) -> TokenStream {
+        if self.is_empty() {
+            return quote! { TypeMeta::Static { size: 0, zero_copy: false } };
+        }
+
+        // Enums have a statically known size in a very specific case: all variants have the same serialized size.
+        // This holds trivially for enums where all variants are unit enums (the size is just the size of the discriminant).
+        // In other cases, we need to compute the size of each variant and check if they are all equal.
+        // Otherwise, the enum is dynamic.
+        //
+        // Enums are never zero-copy, as the discriminant may have invalid bit patterns.
```

Comment:
Invalid bit patterns influence the safety of transmute, but not necessarily whether type is zero-copyable (in the common-sense meaning - I suppose currently in wincode those two concepts are not distinguished), right?

It would also different for write vs read operations:
- write -> serialization is safe (in-mem enum has always correct bit-patterns) and zero-copyable
- read -> deserialization should read some of the bytes to verify valid bits and return Result, but no copying needs to be done

Reply:
Having separate marker for "needs verifycation, but not copying" is probably more of an idea for better optimizations support in future, however given that `SchemaWrite` and `SchemaRead` have their own `TYPE_META`, would it make sense in this case to make enums zero-copyable in `SchemaWrite` and only non-zero-copyable in `SchemaRead`?